subprojects/dependency-management/src/integTest/groovy/org/gradle/integtests/resolve/transform/ArtifactTransformWithDependenciesIntegrationTest.groovy

```diff
@@ -0,0 +1,173 @@
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.gradle.integtests.resolve.transform
+
+import org.gradle.integtests.fixtures.AbstractHttpDependencyResolutionTest
+import org.gradle.util.ToBeImplemented
+
+class ArtifactTransformWithDependenciesIntegrationTest extends AbstractHttpDependencyResolutionTest {
+
+    def setup() {
+        // TODO LJA Investigate, this makes little sense
+        executer.expectDeprecationWarning()
+
+        settingsFile << """
+            rootProject.name = 'transform-deps'
+            include 'lib'
+            include 'app'
+        """
+
+        file('lib/src/main/java/test/MyClass.java') << """
+package test;
+
+public class MyClass {
+    public static void main(String[] args) {
+        System.out.println("Hello world!");
+    }
+}
+"""
+
+        buildFile << """
+def artifactType = Attribute.of('artifactType', String)
+
+allprojects {
+    repositories {
+        mavenCentral()
+    }
+}
+project(':lib') {
+    apply plugin: 'java-library'
+
+    dependencies {
+        api 'org.slf4j:slf4j-api:1.7.25'
+    }
+}
+
+project(':app') {
+    apply plugin: 'java'
+    
+    dependencies {
+        implementation 'junit:junit:4.11'
+        implementation project(':lib')
+    }
+
+    dependencies {
+        registerTransform {
+            from.attribute(artifactType, 'jar')
+            to.attribute(artifactType, 'size')
+            artifactTransform(FileSizer)
+        }
+    }
+}
+
+import javax.inject.Inject
+import org.gradle.api.artifacts.transform.ArtifactTransformDependencies
+
+class FileSizer extends ArtifactTransform {
+
+    ArtifactTransformDependencies artifactDependencies
+
+    @Inject
+    FileSizer(ArtifactTransformDependencies artifactDependencies) {
+        this.artifactDependencies = artifactDependencies
+        println "Creating FileSizer"
+    }
+    
+    List<File> transform(File input) {
+        println "Received dependencies files \${artifactDependencies.dependencies*.name} for processing \${input.name}"
+
+        assert outputDirectory.directory && outputDirectory.list().length == 0
+        def output = new File(outputDirectory, input.name + ".txt")
+        println "Transforming \${input.name} to \${output.name}"
+        output.text = String.valueOf(input.length())
+        return [output]
+    }
+}
+
+"""
+    }
+
+    def "transform can access artifact dependencies as FileCollection when using ArtifactView"() {
```

Comment:
We need a test for chained transformations as well. What dependencies will be provided to transformations further down the chain?

Reply:
It's always the dependencies of the initial artifact.

Reply:
That is what I expected. Can we add a test?

Reply:
Yeah, so that's a problem. We'll need to address this. See main thread of conversation for details.